pennylane/pauli/pauli_arithmetic.py

```diff
@@ -116,6 +116,15 @@ def __init__(self, mapping):
                 del mapping[wire]
         super().__init__(mapping)
 
+    def __reduce__(self):
+        """Defining this method allows PauliWord to be pickle-able. Otherwise, un-pickling
+        would cause __setitem__ to be called, which is forbidden on PauliWord.
+
+        For more information, see: https://docs.python.org/3/library/pickle.html#object.__reduce__
+
+        """
```

Comment:
Maybe we can have a bit more on what exactly `__reduce__` does? It seems the doc-string of this function right now gives more info about a bug it solves and none about what it actually does.

Reply:
imo i'd leave it. this really is _only_ a bugfix, and it doesn't add anything to PL other than that. if you're curious, the link explains why it's needed.

Reply:
As far as I can tell, `__reduce__` is only used by pickling.